soundcard/mediafoundation.py

```diff
@@ -513,17 +517,22 @@ def __init__(self, ptr, samplerate, channels, blocksize, isloopback, exclusive_m
         _com.check_error(hr)
 
         # It's a WAVEFORMATEXTENSIBLE with room for KSDATAFORMAT_SUBTYPE_IEEE_FLOAT:
-        assert ppMixFormat[0][0].Format.wFormatTag == 0xFFFE
-        assert ppMixFormat[0][0].Format.cbSize == 22
-
-        # The data format is float32:
-        # These values were found empirically, and I don't know why they work.
-        # The program crashes if these values are different
-        assert ppMixFormat[0][0].SubFormat.Data1 == 0x100000
-        assert ppMixFormat[0][0].SubFormat.Data2 == 0x0080
-        assert ppMixFormat[0][0].SubFormat.Data3 == 0xaa00
-        assert [int(x) for x in ppMixFormat[0][0].SubFormat.Data4[0:4]] == [0, 56, 155, 113]
-        # the last four bytes seem to vary randomly
+        # Note: Some devices may not return 0xFFFE format, but WASAPI should handle conversion
+        if ppMixFormat[0][0].Format.wFormatTag == 0xFFFE:
+            assert ppMixFormat[0][0].Format.cbSize == 22
+
+            # The data format is float32:
+            # These values were found empirically, and I don't know why they work.
+            # The program crashes if these values are different
+            assert ppMixFormat[0][0].SubFormat.Data1 == 0x100000
+            assert ppMixFormat[0][0].SubFormat.Data2 == 0x0080
+            assert ppMixFormat[0][0].SubFormat.Data3 == 0xaa00
+            assert [int(x) for x in ppMixFormat[0][0].SubFormat.Data4[0:4]] == [0, 56, 155, 113]
+            # the last four bytes seem to vary randomly
+        else:
+            # Device doesn't return WAVEFORMATEXTENSIBLE, but WASAPI will handle conversion
+            # Just skip the assertions and let WASAPI convert
```

Comment:
This looks dubious to me. We probably should at least issue a warning that we're skipping the format checking here.

Unless the whole formatting checks are unnecessary, in which case we can always skip them. But I don't know enough about this API to make that decision, really. Do you know more?